Server/ReasnAPI/ReasnAPI/Controllers/MeController.cs

```diff
@@ -27,72 +27,155 @@ public IActionResult GetCurrentUser()
     }
 
     [HttpPut]
-    public IActionResult UpdateCurrentUser()
+    [ProducesResponseType<UserDto>(StatusCodes.Status200OK)]
+    public IActionResult UpdateCurrentUser(
+        [FromBody] UserDto userDto,
+        [FromServices] IValidator<UserDto> validator)
     {
-        throw new NotImplementedException();
+        validator.ValidateAndThrow(userDto);
+
+        var user = _userService.GetCurrentUser();
+
+        // Non-admin users can't update their role to admin
+        if (user.Role != UserRole.Admin && userDto.Role == UserRole.Admin)
+        {
+            return Forbid();
+        }
+
+        var updatedUser = _userService.UpdateUser(user.Username, userDto);
+
+        return Ok(updatedUser);
     }
 
     [HttpPost]
     [Route("image")]
-    public IActionResult AddCurrentUserImage()
+    [ProducesResponseType<ImageDto>(StatusCodes.Status201Created)]
+    public async Task<IActionResult> AddCurrentUserImage([FromForm] List<IFormFile> images)
     {
-        throw new NotImplementedException();
+        var userId = _userService.GetCurrentUser().Id;
+
+        var imageDtos = new List<ImageDto>();
+
+        foreach (var image in images.Where(image => image.Length > 0))
+        {
+            using var ms = new MemoryStream();
+            await image.CopyToAsync(ms);
+            var fileBytes = ms.ToArray();
+
+            var imageDto = new ImageDto
+            {
+                ObjectId = userId,
+                ObjectType = ObjectType.User,
+                ImageData = fileBytes
+            };
+
+            _imageService.CreateImages([imageDto]);
+            imageDtos.Add(imageDto);
```

Comment:
If we do [imageDto], can't we add dto to list (as you have it) and after for just run createimages for whole list?

Reply:
Users can't upload more than one image, so it's unnecessary to create images from list (`imageDtos.Add(imageDto);` is redundant)